mlflow/utils/rest_utils.py

```diff
@@ -43,10 +44,22 @@ def http_request(host_creds, endpoint, retries=3, retry_interval=3, **kwargs):
 
     verify = not host_creds.ignore_tls_verification
 
+    def request_with_ratelimit_retries(max_rate_limit_interval, **kwargs):
+        response = requests.request(**kwargs)
+        time_left = max_rate_limit_interval
```

Comment:
Do we want to exclude the time spent making the request itself (i.e. the time spent on `requests.request` above) from `time_left`? I remember we'd discussed that 429s should usually be returned quickly, so I'm fine either way, but just wanted to confirm - it could still be nice to do to avoid long user-visible wait times, but on the other hand maybe it's actually good to have the client explicitly sleep between requests to avoid overloading the server (e.g. maybe time spent waiting on the server to respond shouldn't count towards backoff time)

Reply:
Good question. The current semantics of the argument is that it's a limit on how much extra time is the retry logic allowed to add to the request. I think if we included the request time it would be more similar to request timeout. It may be a good idea to have a configurable request timeout but I think that is out of scope of this PR.